packages/test/test-end-to-end-tests/src/test/directoryEndToEndTests.spec.ts

```diff
@@ -1109,7 +1109,7 @@ describeCompat(
 			assert.equal(subDirCreatedEventData[1], "subDirName");
 			assert.equal(undisposedEventData.length, 1);
 			assert.equal(undisposedEventData[0], "/subDirName");
-			assert.equal(disposedEventData.length, 1);
+			assert.equal(disposedEventData.length, 0);
```

Comment:
I changed this e2e test (and the one below) since I think it may have been incorrect. Before this PR we had fired the `disposed` event immediately after `deleteSubDirectory()` was called (assuming there actually was a subdirectory to delete). This PR changes this to only emit the `disposed` event after processing the delete op, which i think is correct but feel free to correct me

Reply:
This was a wrong assumption so in latest we fire `dispose` on local delete